quic/s2n-quic-core/src/recovery/rtt_estimator.rs

```diff
@@ -53,7 +53,14 @@ impl Default for RttEstimator {
 
 impl RttEstimator {
     /// Creates a new RTT Estimator with default initial values using the given `max_ack_delay`.
+    #[inline]
     pub fn new(max_ack_delay: Duration) -> Self {
+        Self::new_with_initial(max_ack_delay, DEFAULT_INITIAL_RTT)
+    }
+
+    /// Creates a new RTT Estimator with the provided initial values using the given `max_ack_delay`.
+    #[inline]
+    pub fn new_with_initial(max_ack_delay: Duration, initial_rtt: Duration) -> Self {
         //= https://www.rfc-editor.org/rfc/rfc9002#section-5.3
```

Comment:
could add something like 
```rust
debug_assert!(initial_rtt >= Duration::from_micros(1));
```
since thats the smallest RTT we allow. Or maybe
```rust
let initial_rtt = initial_rtt.max(Duration::from_micros(1));
```
and I suppose `Duration::from_micros(1)` could be a const too

Reply:
Yeah that's a good point. I think it's probably worth doing both. Once #2041 is addressed, the setter should do the actual validation for this. The check here would then just be a sanity check.